sound/soc/sof/ipc.c

```diff
@@ -424,15 +423,14 @@ static void ipc_trace_message(struct snd_sof_dev *sdev, u32 msg_id)
 static void ipc_period_elapsed(struct snd_sof_dev *sdev, u32 msg_id)
 {
 	struct sof_ipc_stream_posn posn;
+	struct snd_sof_pcm_stream *stream;
 	struct snd_sof_pcm *spcm;
-	u32 posn_offset;
 	int direction;
 
 	/* check if we have stream box */
 	if (sdev->stream_box.size == 0) {
 		/* read back full message */
-		snd_sof_dsp_mailbox_read(sdev, sdev->dsp_box.offset, &posn,
-					 sizeof(posn));
+		snd_sof_ipc_read(sdev, &posn, sizeof(posn));
```

Comment:
I am lost as well on the definition of snd_sof_ipc_read. In one case you use it to read the data part of a RX message, but for trace and period elapsed it's not clear there is a RX message in the first place.

Reply:
snd_sof_ipc_read() is a hw-neutral replacement for snd_sof_dsp_mailbox_read(). So, it is used everywhere where snd_sof_dsp_mailbox_read() was used before.